packages/preact-iso/router.d.ts

```diff
@@ -0,0 +1,19 @@
+import { VNode } from 'preact';
+
+export function LocationProvider(props: { children?: VNode }): VNode;
+
+export function Router(props: { onLoadEnd?: () => void, onLoadStart?: () => void, children?: VNode[] }): VNode;
+
+export const useLoc: () => { url: string, path: string, query: Object, route };
```

Comment:
I think the `query` property is not an Object constructor, but a record. Something like `Record<string, string>`

Reply:
Ah, cheers. I believe this was lifted straight from the JS. Missed it when checking over my work.

Reply:
Do you happen to know what a better type for `route` might be? I'm not too sure of what it's even used for.

Reply:
It looks like it's of the type `(url: string) => void`.

Reply:
Thanks! I appreciate it.